dev-docs/source/CMakeBestPractices.rst

```diff
@@ -0,0 +1,137 @@
+
+If you find yourself doing something like ``target_include_directories(Fitting PUBLIC ${Boost_include_dirs})`` you've probably done something wrong. This goes against the modern CMake target approach. The only thing target_include_directories is needed for is
+to tell users of your Fitting library where it can find its headers:
+
+
+.. code-block:: cmake
+
+    target_include_directories(
+                Fitting PUBLIC
+                $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+    )
+
+This specifies that while building the project, the headers for fitting can be found in the inc/ subfolder.
+
+Tips:
+
+    - Please use this target based approach. Do not link the CMake target to variables such as ``${Boost_LIBRARIES}``.
+    - Consider whether the dependencies are PUBLIC or PRIVATE. If you wrongly specify things as public it add unnecessary dependencies to users of your library.
+    - Please rethink things if you are using ``target_include_directories`` to include headers from an external library, as the ``target_link_libraries`` should have already resolved those requirements.
+
+================================
+Finding a new CMake dependency
+================================
+
+If you are adding a new CMake dependency you'll need to find it using the find_package(...) CMake functions. This looks for the library using two mechanisms:
+
+1. Uses Find(...).cmake either from the CMake inbuilt modules, or using our own ones in the buildconfig/CMake folder.
+2. Uses CMake config files which are optionally installed with the library.
+
+If neither are working, you'll likely need to create your own ``Find(...).cmake`` file. Examples of this include our own ``FindJsoncpp.cmake`` and ``FindPoco.cmake`` files. The primary steps in this file are:
+
+1. Use find_library(MY_LIBRARY NAMES name_of_my_library) to search CMake paths for ``name_of_my_library``
+2. Use find_path(MY_LIBRARY_INCLUDE_DIR header.h)
+3. Finally you should create CMake targets based on these libraries using the IMPORTED specifier.
+
+.. code-block:: cmake
+
+    add_library(MyLibrary::mylibrary UNKNOWN IMPORTED)
+    set_target_properties(
+      MyLibrary::mylibrary
+      PROPERTIES IMPORTED_LOCATION "${MY_LIBRARY NAMES}"
+                 INTERFACE_INCLUDE_DIRECTORIES "${MY_LIBRARY_INCLUDE_DIR}"
+                 IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+    )
+
+After you've created this file, you can find mylibrary and its target ``MyLibrary::mylibrary`` using:
+
+.. code-block:: cmake
+
+    find_package(MyLibrary REQUIRED)
+
+
+NOTE: The ``UNKNOWN`` specifier in add_library is useful on windows as it means we aren't required to point it to the .lib export libraries. The find_library CMake function will not find these. Usage of Unknown is common place in CMake finders because of this.
```

Comment:
```suggestion
NOTE: The ``UNKNOWN`` specifier in add_library is useful on Windows as it means we aren't required to point it to the .lib export libraries. The find_library CMake function will not find these. Usage of Unknown is common place in CMake finders because of this.
```